disperser/batcher/batch_confirmer.go

```diff
@@ -127,6 +132,95 @@ func (b *BatchConfirmer) Start(ctx context.Context) error {
 	return nil
 }
 
+func (b *BatchConfirmer) RecoverState(ctx context.Context) error {
```

Comment:
What keeps this from being just a call to `HandleSingleBatch` that doesn't update the reference block number?